### DISCUSSION TITLE
WrapTextWidth not wrapping/breaking long words

### DISCUSSION THREAD

**Discussion by msipessr** [SixLabors/ImageSharp.Drawing#180] [Q&A] [answered]
I'm trying to wrap text inside a box and it works fine for most text, however if the text is longer then the width set using WrapTextWidth in TextOptions, the word will not wrap (and word break) and it will be drawn outside the box and wrap only the last letter in the word.

My setup:
```c#
TextOptions textOptions = new TextOptions()
   {
         WrapTextWidth = width,
         HorizontalAlignment = SixLabors.Fonts.HorizontalAlignment.Left,
         VerticalAlignment = SixLabors.Fonts.VerticalAlignment.Top
   };

 DrawingOptions drawingOptions = new DrawingOptions()
 {
      TextOptions = textOptions
 };

Image.Mutate(x => x.DrawText(drawingOptions, "Text that will wrap inside the box just fine followed by Areallylongwordthatwontfitinmysetwidth", font, brush, new SixLabors.ImageSharp.Point(xPosition, yPosition)));
```
And the results:

![Screen Shot 2021-12-02 at 1 48 43 PM](https://user-images.githubusercontent.com/95447606/144508597-1fb6df41-a01c-4ca5-92c4-3ce90c92112d.png)

Any idea how to get the long word to break so that it wraps like other text inside the box?  I really don't want to have to manually break apart the word by doing some sort of loop that measures and breaks the word apart and was hoping I'm just missing something somewhere.

**Comment by JimBobSquarePants** [comment, marked as answer]
You'll need to use the latest MyGet build for this. `1.0.0-beta13.15`

There's been some significant changes to the API while we get things ready for RC1.

```c#
using var image = new Image<Rgb24>(404, 147, Color.White);
TextOptions options = new(SystemFonts.CreateFont("arial", 16))
{
    WrappingLength = image.Width * .5F,
    WordBreaking = WordBreaking.BreakAll,
    Origin = new Vector2(20, 20),
};

image.Mutate(x => x.DrawText(options, "Text that will wrap inside the box just fine followed by Areallylongwordthatwontfitinmysetwidth", Color.Black));
image.SaveAsPng("word-break.png");
```
![word-break](https://user-images.githubusercontent.com/385879/144515545-25921cdf-203c-4431-acb3-1f932d8d6793.png)

**Reply by msipessr** [reply]
That worked perfectly!  Thank you!  I've been banging my head against this for a couple of days now.

**Reply by JimBobSquarePants** [reply]
No worries, happy to help!
